quail/processes/wps_climdex_mmdmt.py

```diff
@@ -0,0 +1,148 @@
+import os
+from rpy2 import robjects
+from pywps import Process, LiteralInput
+from pywps.app.Common import Metadata
+from pywps.app.exceptions import ProcessError
+
+from wps_tools.utils import log_handler, collect_args, common_status_percentages
+from wps_tools.io import log_level
+from quail.utils import get_package, logger, load_rdata_to_python, save_python_to_rdata
+from quail.io import climdex_input, ci_name, output_file, rda_output, vector_name
+
+
+class ClimdexMMDMT(Process):
+    """
+    This process wraps climdex function
+    - climdex.txx: Monthly (or annual) Maximum of Daily Maximum Temperature
+    """
+
+    def __init__(self):
+        self.status_percentage_steps = dict(
+            common_status_percentages,
+            **{
+                "load_rdata": 10,
+                "save_rdata": 90,
+            },
+        )
+        inputs = [
+            climdex_input,
+            ci_name,
+            output_file,
+            LiteralInput(
+                "month_type",
+                "Month type to compute",
+                abstract="Min/ max daily temperature type to compute",
+                allowed_values=["txx"],
+                min_occurs=1,
+                max_occurs=1,
+                data_type="string",
+            ),
+            LiteralInput(
+                "freq",
+                "Frequency",
+                abstract="Time frequency to aggregate to",
+                allowed_values=["monthly", "annual"],
+                default="monthly",
+                min_occurs=0,
+                max_occurs=1,
+                data_type="string",
+            ),
+            vector_name,
+            log_level,
+        ]
+
+        outputs = [rda_output]
+
+        super(ClimdexMMDMT, self).__init__(
+            self._handler,
+            identifier="climdex_mmdmt",
+            title="Climdex MMDMT",
+            abstract="Monthly or Annual Maximum of Daily Maximum Temperature",
+            metadata=[
+                Metadata("NetCDF processing"),
+                Metadata("Climate Data Operations"),
+                Metadata("PyWPS", "https://pywps.org/"),
+                Metadata("Birdhouse", "http://bird-house.github.io/"),
+                Metadata("PyWPS Demo", "https://pywps-demo.readthedocs.io/en/latest/"),
+            ],
+            inputs=inputs,
+            outputs=outputs,
+            store_supported=True,
+            status_supported=True,
+        )
+
+    def MMDMT_type(self, month_type, ci, freq):
+        climdex = get_package("climdex.pcic")
+
+        if month_type == "txx":
+            return climdex.climdex_txx(ci, freq)
```

Comment:
Also, I don't really understand the use of the input `month_type` reading process. How is this input used for `climdex.txx` method?

Reply:
It is like the same as [days_type](https://github.com/pacificclimate/quail/blob/4be46d69ad71ddd95378d8912ac1e77dc6dbbf62/quail/processes/wps_climdex_days.py#L90) in `wps_climdex_days` because I will be adding more functions to wrap